OSDOCS-3799 External DNS install CLI instructions (#76037)
openshift/openshift-docs: modules/nw-installing-external-dns-operator-cli.adoc

+
[source,terminal]
----
$ oc -n external-dns-operator \
    get ip <install_plan_name> \
    --template='{{.status.phase}}{{"\n"}}'
----

. Verify that the status of the `external-dns-operator` pod is `Running` by running the following command:
+
[source,terminal]
----
$ oc -n external-dns-operator get pod
----

+
.Example output
[source,terminal]
----
NAME                                     READY   STATUS    RESTARTS   AGE
external-dns-operator-5584585fd7-5lwqm   2/2     Running   0          11m
----

. Verify that the catalog source of the subscription is `redhat-operators` by running the following command:
+
[source,terminal]
----
$ oc -n external-dns-operator get subscription
----

+
.Example output
[source,terminal]
----
NAME                    PACKAGE                 SOURCE             CHANNEL
external-dns-operator   external-dns-operator   redhat-operators   stable-v1
----

. Check the `external-dns-operator` version by running the following command:
+
[source,terminal]
----
$ oc -n external-dns-operator get csv
----

+
.Example output
[source,terminal]
----
NAME                           DISPLAY                VERSION   REPLACES   PHASE

> external-dns-operator.v<1.y.z>   ExternalDNS Operator   <1.y.z><1>                Succeeded

Reviewer: One teeny thing on preview, I notice that this callout isn't rendering. You may need to play with this, but I _think_ putting a space, ie >space here<1> will work. 

![image](https://github.com/user-attachments/assets/cd8d48ab-454e-4323-8ad4-c782d447bd6c)

Author: TYSM

> NAME                           DISPLAY                VERSION   REPLACES   PHASE
> external-dns-operator.v<1.y.z>   ExternalDNS Operator   <1.y.z> <1>                Succeeded
> ----
> <1>The 1.y.z variable specifies the External DNS Operator version that you installed. Replace 1.y.z with your External DNs Operator version.

Reviewer: ```suggestion
NAME                           DISPLAY                VERSION   REPLACES   PHASE
external-dns-operator.v<1.y.z>   ExternalDNS Operator   <1.y.z>   Succeeded
----
+
The 1.y.z variable specifies the External DNS Operator version that you installed. Replace 1.y.z with your External DNs Operator version.
```

Reviewer: Looks like that didn't work afterall. In this case, I wonder if its just the placement of the callout, not being the last item here, that's making it not render 🤔 I think that you could probably get away with no callout here, just add a `+` as I indicate in the suggestion to keep the explanatory text inline.

Author: @skrthomas I did away with the explanatory text as it isn't required.

// Module included in the following assemblies:
//
// * networking/external_dns_operator/nw-installing-external-dns-operator-on-cloud-providers.adoc

:_mod-docs-content-type: PROCEDURE
[id="nw-installing-external-dns-operator-cli_{context}"]
= Installing the External DNS Operator by using the CLI

You can install the External DNS Operator by using the CLI.

.Prerequisites

* You are logged in to the {product-title} web console as a user with `cluster-admin` permissions.

* You are logged into the OpenShift CLI (`oc`).

.Procedure

. Create a `Namespace` object:

.. Create a YAML file that defines the `Namespace` object:
+
.Example `namespace.yaml` file
[source,yaml]
----
apiVersion: v1
kind: Namespace
metadata:
  name: external-dns-operator
----

.. Create the `Namespace` object by running the following command:
+
[source,terminal]
----
$ oc apply -f namespace.yaml
----

. Create an `OperatorGroup` object:

.. Create a YAML file that defines the `OperatorGroup` object:
+
.Example `operatorgroup.yaml` file
[source,yaml]
----
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:
  name: external-dns-operator
  namespace: external-dns-operator
spec:

>   upgradeStrategy: Default
>     targetNamespaces:
>   - external-dns-operator

Reviewer 2: ```suggestion
  upgradeStrategy: Default
  targetNamespaces:
  - external-dns-operator
```

Author: ty

Reviewer 2: This one is still having the leading whitespace.

----

.. Create the `OperatorGroup` object by running the following command:
+
[source,terminal]
----
$ oc apply -f operatorgroup.yaml
----

. Create a `Subscription` object:

.. Create a YAML file that defines the `Subscription` object:
+
.Example `subscription.yaml` file
[source,yaml]
----
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: external-dns-operator
  namespace: external-dns-operator
spec:
  channel: stable-v1
  installPlanApproval: Automatic
  name: external-dns-operator
  source: redhat-operators
  sourceNamespace: openshift-marketplace
----

.. Create the `Subscription` object by running the following command:
+
[source,terminal]
----
$ oc apply -f subscription.yaml
----

.Verification

. Get the name of the install plan from the subscription:
+
[source,terminal]
----
$ oc -n external-dns-operator \
    get subscription external-dns-operator \
    --template='{{.status.installplan.name}}{{"\n"}}'
----

> . Check the status of the install plan:

Author: ty

+
[source,terminal]
----
$ oc -n external-dns-operator \
    get ip <install_plan_name> \
    --template='{{.status.phase}}{{"\n"}}'
----
+
.Required output
[source,terminal]
----
Complete

> ----

Author: ty

> . Check the status of the `external-dns-operator` pod is running:

Reviewer 3: ```suggestion
. Verify that the status of the `external-dns-operator` pod is `Running` by running the following command:
```
- Same comment about using "verify" instead of "check"
- Same comment using "by running the following command"
- The status of the pod is shown as "Running" with capital R.
- A pod status should be in back ticks.

Author: TY

+
[source,terminal]
----
$ oc -n external-dns-operator get pod
----

+
.Example output
[source,terminal]
----
NAME                                     READY   STATUS    RESTARTS   AGE
external-dns-operator-5584585fd7-5lwqm   2/2     Running   0          11m
----

> . Confirm the subscription is `Red Hat Operators` catalog source:

Reviewer 3: ```suggestion
. Confirm that the catalog source of the subscription is `redhat-operators` by running the following command:
```

- Per the command output, the value in back ticks should be `redhat-operators`, right?
- Same comment to use "by running the..."

Author: I agree with the first comment but I think the second comment is unnecessary clutter, as it's obvious that a command follows the direction.

+
[source,terminal]
----
$ oc -n external-dns-operator get subscription

> ----

Reviewer 2: I think this verification redundant, the catalog source was set in `subscription.yaml`.

+
.Example output
[source,terminal]
----
NAME                    PACKAGE                 SOURCE             CHANNEL
external-dns-operator   external-dns-operator   redhat-operators   stable-v1
----

> . Check the `external-dns-operator` version:

Reviewer 3: ```suggestion
. Check the External DNS Operator version by running the following command:
```
- Is this step verifying that the External DNS Operator version is updated to a correct value? If yes, then same comment about using "verify".
- IMO, instead of using a resource name `external-dns-operator`, how about using the Operator name?
- I don't think this command always expects the EDNS Operator version to be `1.2.0` after installing the Operator. (SME can confirm though). So considering that the version changes for every Operator release, I would leave up to you to use a variable for specifying the version.

Author: ty

+
[source,terminal]
----
$ oc -n external-dns-operator get csv
----

+
.Example output
[source,terminal]

> ----
> NAME                           DISPLAY                VERSION   REPLACES   PHASE
> external-dns-operator.v1.2.0   ExternalDNS Operator   1.2.0                Succeeded
> ----

Reviewer 3: ```suggestion
----
NAME                           DISPLAY                VERSION   REPLACES   PHASE
external-dns-operator.v<1.y.z>   ExternalDNS Operator   <1.y.z>                Succeeded
----
```

- You can add a call out in the output and something similar to:

 > <1> The `<1.y.z>` variable specifies      the External DNS Operator version that you installed, for example, `1.2.0`. Replace `<1.y.z>` with your External DNs Operator version.

- If you go with this suggestion, you can confirm with SME if `1.y.z` is appropriate. (Like in OCP we have 4.y.z ) [micro release guideline](https://redhat-documentation.github.io/supplementary-style-guide/#micro-release)

Author: Thanks, I'll take a stab at it.

// Module included in the following assemblies:
//
// * networking/external_dns_operator/nw-installing-external-dns-operator-on-cloud-providers.adoc

:_mod-docs-content-type: PROCEDURE
[id="nw-installing-external-dns-operator-cli_{context}"]
= Installing the External DNS Operator by using the CLI

You can install the External DNS Operator by using the CLI.

.Prerequisites

* You are logged in to the {product-title} web console as a user with `cluster-admin` permissions.

* You are logged into the OpenShift CLI (`oc`).

.Procedure

. Create an `OperatorGroup` object:

.. Create a YAML file that defines the `OperatorGroup` object:
+

> .Example `operatorgroup.yaml` file

Reviewer 2: There is one step missing - the creation of the `external-dns-operator` namespace. It should be the first step. You can see how it's done in AWS LB Operator module.

[source,yaml]
----
apiVersion: operators.coreos.com/v1
kind: OperatorGroup
metadata:

>   name: aws-lb-operatorgroup

Reviewer 2: ```suggestion
  name: external-dns-operator
```

  namespace: external-dns-operator
spec:

>   upgradeStrategy: Default
> ----

Reviewer 2: We want to install only in `external-dns-operator` namespace.

```suggestion
  upgradeStrategy: Default
  targetNamespaces:
  - external-dns-operator
----
```

.. Create the `OperatorGroup` object by running the following command:
+
[source,terminal]
----
$ oc apply -f operatorgroup.yaml
----

. Create a `Subscription` object:

.. Create a YAML file that defines the `Subscription` object:
+
.Example `subscription.yaml` file
[source,yaml]
----
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  name: external-dns-operator
  namespace: external-dns-operator
spec:
  channel: stable-v1
  installPlanApproval: Automatic
  name: external-dns-operator

>   source: qe-app-registry

Reviewer 2: ```suggestion
  source: redhat-operators
```

Btw I saw the same catalog source in ALBO. It should be the official Red Hat operators catalog source, not QE.
